engine/src/uci/optionsuci.cpp

```diff
@@ -217,20 +218,43 @@ void OptionsUCI::setoption(istringstream &is, Variant& variant, StateObj& state)
         }
 #endif
         Options[name] = value;
-        cout << "info string Updated option " << givenName << " to " << value << endl;
-        if (name == "uci_variant") {
+        if (name != "uci_variant") {
+            cout << "info string Updated option " << givenName << " to " << value << endl;
+        } else {
 #ifdef XIANGQI
             // Workaround. Fairy-Stockfish does not use an enum for variants
             cout << "info string variant " << "Xiangqi" << " startpos " << "rnbakabnr/9/1c5c1/p1p1p1p1p/9/9/P1P1P1P1P/1C5C1/9/RNBAKABNR w - - 0 1" << endl;
 #else
+            if (value == "standard") {
+                Options["UCI_Variant"] << Option("chess");
+            }
+#ifdef MODE_LICHESS
+            if (value == "antichess" || value == "losers") {
+                Options["UCI_Variant"] << Option("giveaway");
+            }
+#endif // MODE_LICHESS
+            string suffix_960 = "";
+#ifdef SUPPORT960
+            if (value == "fischerandom" || value == "chess960"
+                || ((value == "chess") && Options["UCI_Chess960"])) {
+                Options["UCI_Variant"] << Option("chess");
+                Options["UCI_Chess960"] << Option(true);
+                cout << "info string Updated option " << givenName << " to chess" << endl;
+                cout << "info string Updated option uci_chess960 to true" << endl;
+                suffix_960 = "960";
+            } else {
+                Options["UCI_Chess960"] << Option(false); // We only want 960 for chess at the moment
+            }
+#endif // SUPPORT960
+            if(suffix_960 == "") {
+                cout << "info string Updated option " << givenName << " to " << value << endl;
+            }
+            Options["Model_Directory"] << Option(((string) "model/" + string(Options["UCI_Variant"]) + suffix_960).c_str());
+            Options["Model_Directory_Contender"] << Option(((string) "model_contender/" + string(Options["UCI_Variant"]) + suffix_960).c_str());
```

Comment:
Maybe it makes sense to move parts of this code into a new method e.g. to handle all alternative input names for variants.

Reply:
Done! I added a function which handles the uci names.